### ISSUE TITLE
[Question/Suggestion] Pass data of an output channel to the next one

### ISSUE THREAD

**Issue by alexej-d** [dachcom-digital/pimcore-formbuilder#486]
| Q                | A      |
|------------------|--------|
| Bug report?      | no |
| Feature request? | yes |
| BC Break report? | no |
| RFC?             | no |

I was wondering if it is possible to pass some of an output channel's returned data to the next channel in the same workflow?

Imagine you have a custom output channel that creates a PDF and you want to pass the URL of the PDF to the Email channel.

How would you do that? Maybe via a global service that stores the current lifecycle's context variables?

**Comment by alexej-d** [comment]
Any recommendations @solverat ?

**Comment by solverat** [comment]
Hey @alexej-d Sorry for the late response, but I need(ed) some time to think about this. 

tl;dr: No. Currently, there is no way to fetch data between the channel processes.

We have an internal signal listener/workflow, which is not dispatching after every channel process. It's also not meant to store dynamic data.

However, I just pushed something here:

https://github.com/dachcom-digital/pimcore-formbuilder/pull/504

This will introduce a new object called `ChannelContext`. It's available in the whole workflow lifecycle and can be retrieved in every `ChannelSubjectGuardEvent` event or within your custom channel, if you're implementing the `ChannelContextAwareInterface` interface.

With this, you should be able to pass the PDF to every upcoming channel.

**Comment by solverat** [comment]
Refs:
- https://github.com/dachcom-digital/pimcore-formbuilder/blob/236e8b97232506c774a055b0dce304ed2771d890/docs/OutputWorkflow/13_ChannelContext.md
-  https://github.com/dachcom-digital/pimcore-formbuilder/blob/1ed307c0e6e7d1607dc692bfa26c1a496e6f61a2/docs/OutputWorkflow/30_Events.md

**Comment by alexej-d** [comment]
Thank you @solverat! Will play with the new addition ASAP!
